docs/webactions.md

```diff
@@ -0,0 +1,133 @@
+# Web Actions (Experimental)
+
+Web actions are OpenWhisk actions annotated to quickly enable you to build web based applications. This allows you to program backend logic which your web application can access anonymously without requiring an OpenWhisk authentication key. It is up to the action developer to implement their own desired authentication and authorization (i.e. OAuth flow).
+
+Web action activations will be associated with the user that created the action. This actions defers the cost of an action activation from the caller to the owner of the action. **Note**: This feature is currently an experimental offering that enables users an early opportunity to try it out and provide feedback.
+
+Let's take the following JavaScript action `hello.js`,
+```javascript
+$ cat hello.js
+function main({name}) {
+  var msg = 'you did not tell me who you are.';
+  if (name) {
+    msg = `hello ${name}!`
+  }
+  return {body: `<html><body><h3>${msg}</h3></body></html>`}
+}
+```
+
+You may create a _web action_ `hello` in the package `demo` for the namespace `guest` using the annotation `web-export`:
+```bash
+$ wsk package create demo
+$ wsk action create /guest/demo/hello hello.js -a web-export true
+```
+
+The `web-export` annotation allows the action to be accessible as a web action via a new REST interface. The URL that is structured as follows: `https://{APIHOST}/api/v1/experimental/web/{QUALIFIED ACTION NAME}.{EXT}`. The fully qualified name of an action consists of three parts: the namespace, the package name, and the action name. An example is `guest/demo/hello`. The last part of the URI called the `extension` which is typically `.http` although other values are permitted as described later. The web action API path may be used with `curl` or `wget` without an API key. It may even be entered directly in your browser.
+
+Try opening [https://${APIHOST}/api/v1/experimental/web/guest/demo/hello.http?name=Jane](https://${APIHOST}/api/v1/experimental/web/guest/demo/hello.http?name=Jane) in your web browser. Or try invoking the action via `curl`:
+```
+$ curl https://${APIHOST}/api/v1/experimental/web/guest/demo/hello.http?name=Jane
+```
+
+Here is an example of a web action that performs an HTTP redirect:
+```javascript
+function main() {
+  return { 
+    headers: { "location": "http://openwhisk.org" }, 
+    code: 302 
+  }
+}
+```  
+
+Or sets a cookie:
+```javascript
+function main() {
+  return { 
+    headers: { 
+      "Set-Cookie": "UserID=Jane; Max-Age=3600; Version=",
+      "Content-Type": "text/html"  
+    }, 
+    code: 200, 
+    body: "<html><body><h3>hello</h3></body></html" }
+}
+```
+
+Or returns an `image/png`:
+```
+function main() {
+    let png = <base 64 encoded string>
+    return { headers: { 'Content-Type': 'image/png' },
+             code: 200,
+             body: png };
+}
+```
+
```

Comment:
This are outdated comments, see below the example of `application/json` it's already in the PR